rllib/core/learner/learner_group.py

```diff
@@ -231,43 +239,77 @@ def _distributed_update(
             See `.update()` docstring.
 
         Returns:
-            A list of dictionaries of results from the updates from the Learner(s)
+            A list of dictionaries of results from the updates from the individual
+            Learner(s)
         """
+        # Make sure minibatch size is reduced to the correct number of shards as well
+        # (just like we split each batch into the number of learner workers).
+        if minibatch_size is not None:
+            minibatch_size //= len(self._workers)
+
+        def _learner_update(learner, minibatch):
```

Comment:
Using a partial now, not lambda to avoid late binding of outer variable `minibatch` (closures).